cwms_radar_api/src/main/java/cwms/radar/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -0,0 +1,649 @@
+	{
+		List<RecentValue> retval  = new ArrayList<>();
+
+		if (categoryId != null && groupId != null) {
+			AV_TSV_DQU tsvView = AV_TSV_DQU.AV_TSV_DQU;  // should we look at the daterange and possible use 30D view?
+
+			AV_TS_GRP_ASSGN tsView = AV_TS_GRP_ASSGN.AV_TS_GRP_ASSGN;
+
+			SelectConditionStep<Record> innerSelect
+					= dsl.select(tsvView.asterisk(), tsView.TS_ID, tsView.ATTRIBUTE,
+					max(tsvView.DATE_TIME).over(partitionBy(tsvView.TS_CODE)).as("max_date_time"), tsView.TS_ID)
+					.from(tsvView.join(tsView).on(tsvView.TS_CODE.eq(tsView.TS_CODE.cast(Long.class))))
+					.where(
+							tsView.DB_OFFICE_ID.eq(office)
+									.and(tsView.CATEGORY_ID.eq(categoryId))
+									.and(tsView.GROUP_ID.eq(groupId))
+									.and(tsvView.VALUE.isNotNull())
+									.and(tsvView.DATE_TIME.lt(futureLimit))
+									.and(tsvView.DATE_TIME.gt(pastLimit))
+									.and(tsvView.START_DATE.le(futureLimit))
+									.and(tsvView.END_DATE.gt(pastLimit)));
+
+			Field[] queryFields = new Field[]{
+					tsvView.OFFICE_ID,
+					tsvView.TS_CODE,
+					tsvView.DATE_TIME,
+					tsvView.VERSION_DATE,
+					tsvView.DATA_ENTRY_DATE,
+					tsvView.VALUE,
+					tsvView.QUALITY_CODE,
+					tsvView.START_DATE,
+					tsvView.END_DATE,
+					tsvView.UNIT_ID,
+					tsView.TS_ID, tsView.ATTRIBUTE};
+
+			List<Field<Object>> fields = Arrays.stream(queryFields)
+					.map(Field::getName)
+					.map(DSL::field).collect(
+					Collectors.toList());
+
+
+			// I want to select tsvView.asterisk but we are selecting from an inner select and
+			// even though the inner select selects tsvView.asterisk it isn't the same.
+			// So we will just select the fields we want.
+			// Unfortunately that means our results won't map into AV_TSV.AV_TSV
+			dsl.select(fields)
+					.from(innerSelect)
+					.where(field(tsvView.DATE_TIME.getName()).eq(innerSelect.field("max_date_time")))
+					.orderBy(field(tsView.ATTRIBUTE.getName()))
+					.forEach( jrecord -> {
+						RecentValue recentValue = buildRecentValue(tsvView, tsView, jrecord);
+						retval.add(recentValue);
+					});
+		}
+		return retval;
+	}
+
+	@NotNull
+	private TsvDquId buildDquId(AV_TSV_DQU tsvView, Record jrecord)
+	{
+		return new TsvDquId(jrecord.getValue(tsvView.OFFICE_ID.getName(), String.class),
+				jrecord.getValue(tsvView.TS_CODE.getName(), Long.class),
+				jrecord.getValue(tsvView.UNIT_ID.getName(), String.class),
+				jrecord.getValue(tsvView.DATE_TIME.getName(), Timestamp.class));
+	}
+
+	public void create(TimeSeries input)
+	{
+		dsl.connection(connection -> {
+			CwmsDbTs tsDao = CwmsDbServiceLookup.buildCwmsDb(CwmsDbTs.class, connection);
+
+			int utcOffsetMinutes = 0;
+			int intervalForward = 0;
+			int intervalBackward = 0;
+			boolean versionedFlag = false;
+			boolean activeFlag = true;
+			tsDao.createTsCodeBigInteger(connection, input.getOfficeId(), input.getName(), utcOffsetMinutes,
```

Comment:
are we sure this function success it's definitely worked? If so perhaps we should catch the exception on error and return our own indication. This is one of those places were a non-generic error message may be handy.